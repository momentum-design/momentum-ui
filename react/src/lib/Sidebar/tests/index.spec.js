import React from 'react';
import { shallow, mount } from 'enzyme';
import {
  Sidebar,
  SidebarNav,
  SidebarNavItem,
} from '@momentum-ui/react';

describe('<Sidebar />', () => {
  it('should render a Sidebar', () => {
    const wrapper = shallow(<Sidebar />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should add customized class name if className prop is set', () => {
    const wrapper = shallow(<Sidebar className='testClassName'/>);

    expect(wrapper.find('.testClassName').exists()).toBeTruthy();
  });

  it('should add customized wrapper class name if wrapperClass prop is set', () => {
    const wrapper = shallow(<Sidebar wrapperClass='testClassName'/>);

    expect(wrapper.find('.testClassName').exists()).toBeTruthy();
  });

  it('should render children if children prop is set', () => {
    const wrapper = shallow(
      <Sidebar>
        <div className='sample-children'>Sample Children</div>
      </Sidebar>
    );

    expect(wrapper.find('.sample-children').exists()).toBeTruthy();
  });

  it('should add customized theme name if theme prop is set', () => {
    const wrapper = shallow(<Sidebar theme='dark'/>);

    expect(wrapper.find('.md-sidebar--dark').exists()).toBeTruthy();
  });

  it('should minimize Sidebar if the collapsable/!expanded/withIcons are set', () => {
    const wrapper = mount(<Sidebar collapsable expanded={false} />);

    expect(wrapper.find('.md-sidebar--minimized').exists()).toBeTruthy();
  });

  it('should collapse Sidebar if the collapsable/!expanded/!withIcons', () => {
    const wrapper = mount(<Sidebar collapsable expanded={false} withIcons={false} />);

    expect(wrapper.find('.md-sidebar--collapsed').exists()).toBeTruthy();
  });

  it('should show Sidebar if the collapsable and expanded are set', () => {
    const wrapper = shallow(<Sidebar collapsable expanded />);

    expect(wrapper.find('.md-sidebar--collapsed').exists()).toBeFalsy();
  });

  it('should add fixed if isFixed prop is set', () => {
    const wrapper = shallow(<Sidebar isFixed/>);

    expect(wrapper.find('.md-sidebar__wrapper--fixed').exists()).toBeTruthy();
    expect(wrapper.find('.md-sidebar--fixed').exists()).toBeTruthy();
  });

  it('should not add nested class if it has icons and no tiers in children', () => {
    const wrapper = shallow(<Sidebar isFixed/>);

    expect(wrapper.find('.md-sidebar--nested').exists()).toBeFalsy();
  });

  it('should add nested class if it has icons and secondary tiers in children', () => {
    const wrapper = shallow(<Sidebar isFixed/>);
    wrapper.setState({
      sidebarContext: {
        secondary: true
      }
    });

    expect(wrapper.find('.md-sidebar--nested').exists()).toBeTruthy();
  });

  it('should add nested class if it has icons and secondary tiers in children', () => {
    const wrapper = shallow(<Sidebar isFixed/>);
    wrapper.setState({
      sidebarContext: {
        tertiary: true
      }
    });

    expect(wrapper.find('.md-sidebar--nested').exists()).toBeTruthy();
  });

  it('should not have global class with isPageLevel true', () => {
    const wrapper = shallow(<Sidebar isPageLevel/>);

    expect(wrapper.find('.md-sidebar--global').exists()).toBeFalsy();
  });

  it('should add indented class if withIcons is true', () => {
    const wrapper = shallow(<Sidebar withIcons/>);

    expect(wrapper.find('.md-sidebar--indented').exists()).toBeTruthy();
  });

  it('should add indented class if withIcons=(true), isPageLevel=(true)', () => {
    const wrapper = shallow(<Sidebar withIcons isPageLevel/>);

    expect(wrapper.find('.md-sidebar--indented').exists()).toBeFalsy();
  });

  it('should add topbar class if withTopbar is true', () => {
    const wrapper = shallow(<Sidebar withTopbar/>);

    expect(wrapper.find('.md-sidebar--topbar').exists()).toBeTruthy();
  });

  it('should have nested class if nested SidebarItems', () => {
    const wrapper = mount(
      <Sidebar>
        <SidebarNav>
          <SidebarNavItem>
            <SidebarNavItem />
          </SidebarNavItem>
        </SidebarNav>
      </Sidebar>
    );

    expect(wrapper.find('.md-sidebar--nested').exists()).toBeTruthy();
  });

  it('should add toggle button if withToggle is true', () => {
    const wrapper = shallow(<Sidebar withToggle />);

    expect(wrapper.find('.md-sidebar__toggle').exists()).toBeTruthy();
  });

  it('should not add toggle button if withToggle is false', () => {
    const wrapper = shallow(<Sidebar />);

    expect(wrapper.find('.md-sidebar__toggle').exists()).toBeFalsy();
  });

  it('should add minimized toggle button if withIcons/withToggle/!expanded set', () => {
    const wrapper = mount(<Sidebar withIcons withToggle expanded={false}/>);

    expect(wrapper.find('.md-sidebar__toggle--minimized').exists()).toBeTruthy();
  });

  it('should add collapsed toggle button if !withIcons/withToggle/!expanded set', () => {
    const wrapper = mount(<Sidebar withIcons={false} withToggle expanded={false}/>);

    expect(wrapper.find('.md-sidebar__toggle--collapsed').exists()).toBeTruthy();
  });

  it('should apply buttonProps to toggle button', () => {
    const wrapper = mount(<Sidebar withToggle buttonProps={{ ariaLabel: 'test' }}/>);

    expect(wrapper.find('.md-button').props()['aria-label']).toEqual('test');
  });
});
