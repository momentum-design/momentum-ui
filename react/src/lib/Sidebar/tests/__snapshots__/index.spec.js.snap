// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Sidebar /> should render a Sidebar 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Sidebar
    buttonProps={Object {}}
    className=""
    collapsable={false}
    expanded={true}
    isFixed={false}
    isPageLevel={false}
    theme=""
    withIcons={true}
    withToggle={false}
    withTopbar={false}
    wrapperClass=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="md-sidebar md-sidebar--global md-sidebar--indented"
        >
          <Context.Provider
            value={
              Object {
                "primary": true,
                "secondary": false,
                "setContext": [Function],
                "tertiary": false,
              }
            }
          >
            <UIDReset />
          </Context.Provider>
        </div>,
        false,
      ],
      "className": "md-sidebar__wrapper",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Context.Provider
            value={
              Object {
                "primary": true,
                "secondary": false,
                "setContext": [Function],
                "tertiary": false,
              }
            }
          >
            <UIDReset />
          </Context.Provider>,
          "className": "md-sidebar md-sidebar--global md-sidebar--indented",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <UIDReset />,
            "value": Object {
              "primary": true,
              "secondary": false,
              "setContext": [Function],
              "tertiary": false,
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": null,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": Object {
            "$$typeof": Symbol(react.provider),
            "_context": Object {
              "$$typeof": Symbol(react.context),
              "Consumer": Object {
                "$$typeof": Symbol(react.context),
                "_calculateChangedBits": null,
                "_context": [Circular],
              },
              "Provider": [Circular],
              "_calculateChangedBits": null,
              "_currentRenderer": null,
              "_currentRenderer2": null,
              "_currentValue": null,
              "_currentValue2": null,
              "_threadCount": 0,
            },
          },
        },
        "type": "div",
      },
      false,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="md-sidebar md-sidebar--global md-sidebar--indented"
          >
            <Context.Provider
              value={
                Object {
                  "primary": true,
                  "secondary": false,
                  "setContext": [Function],
                  "tertiary": false,
                }
              }
            >
              <UIDReset />
            </Context.Provider>
          </div>,
          false,
        ],
        "className": "md-sidebar__wrapper",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Context.Provider
              value={
                Object {
                  "primary": true,
                  "secondary": false,
                  "setContext": [Function],
                  "tertiary": false,
                }
              }
            >
              <UIDReset />
            </Context.Provider>,
            "className": "md-sidebar md-sidebar--global md-sidebar--indented",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <UIDReset />,
              "value": Object {
                "primary": true,
                "secondary": false,
                "setContext": [Function],
                "tertiary": false,
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": null,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": Object {
              "$$typeof": Symbol(react.provider),
              "_context": Object {
                "$$typeof": Symbol(react.context),
                "Consumer": Object {
                  "$$typeof": Symbol(react.context),
                  "_calculateChangedBits": null,
                  "_context": [Circular],
                },
                "Provider": [Circular],
                "_calculateChangedBits": null,
                "_currentRenderer": null,
                "_currentRenderer2": null,
                "_currentValue": null,
                "_currentValue2": null,
                "_threadCount": 0,
              },
            },
          },
          "type": "div",
        },
        false,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
